JsonSchema/JsonSchema.cs

```diff
@@ -405,7 +407,7 @@ private static SpecVersion DetermineSpecVersion(JsonSchema schema, SchemaRegistr
 	{
 		if (schema.BoolValue.HasValue) return SpecVersion.DraftNext;
 		if (schema.DeclaredVersion != SpecVersion.Unspecified) return schema.DeclaredVersion;
-		if (!Enum.IsDefined(typeof(SpecVersion), desiredDraft)) return desiredDraft;
```

Comment:
I didn't quite understand the intention here, so I hope I got it right.

Reply:
This actually supports an external library, my Graeae OpenAPI lib, by allowing it to define draft 4 as an enum value.  It's an undocumented hack, really, but it needs to work.